source/base/function_signed_distance.cc

```diff
@@ -332,6 +332,98 @@ namespace Functions
 
       return evaluate_ellipsoid(point) < 0.0 ? -distance : distance;
     }
+
+
+
+    template <int dim>
+    Rectangle<dim>::Rectangle(const Point<dim> &bottom_left,
+                              const Point<dim> &top_right)
+      : bounding_box({bottom_left, top_right})
+    {}
+
+
+
+    template <int dim>
+    double
+    Rectangle<dim>::value(const Point<dim> & p,
+                          const unsigned int component) const
+    {
+      (void)component;
+
+      // calculate distance vector to bounding box
```

Comment:
@mschreter One favor. I think the code below fits well into `BoundingBox`; something like `BoundigBox::distance(Point)`. Would it be possible to move these lines there and call the new function here?

Reply:
Mmh, I personally would expect from a function `distance` a scalar value.

Reply:
`signed_distance` sounds good :+1: The code above also computes a `scalar`; I would move the whole content and not only the first few lines!

Reply:
Something like this:

```cpp
template <int spacedim, typename Number>
Number
BoundingBox<spacedim, Number>::signed_distance(const Point<spacedim, Number>  &point,
                                        const unsigned int direction) const
{
  const Number p1 = lower_bound(direction);
  const Number p2 = upper_bound(direction);

  if (point[direction] > p2)
    return point[direction] - p2;
  else if (point[direction] < p1)
    return p1 - point[direction];
  else
    return -std::min(point[direction] - p1, p2 - point[direction]);
}
```

Reply:
I see. What about creating a second function that only takes a point and calls the function suggested by you?